custom-images/generate_custom_image.py

```diff
@@ -48,6 +48,7 @@
 # New style images: 1.2.3-deb8
 _VERSION_REGEX = re.compile(r"^\d+\.\d+\.\d+(-.{4})?$")
 _IMAGE_URI = "projects/cloud-dataproc/global/images/{}"
+_PARTIAL_IMAGE_URI = re.compile(r"^projects\/.+\/global\/images\/.+$")
```

Comment:
nit: its all too easy to copy/paste selfLink which is always a full URI. Consider making this: (https://..../)?projects/...